Sage/carbon
# fix(date): date formatting when locale overriden (FE-4391)

### Proposed behaviour
Include moment locales to allow user to use their own locale.
Remove date "sanitization" in `_parseDate` method to enforce ISO date in the `value` prop, which should also fix all day/month swap issues.
Modify custom event generation so the event passed to the `onBlur` contains correct values when the date is correct.



### Current behaviour
Localization for the date component is not working as an error is triggered when a user is trying to implement it.
Date is being incorrectly validated, which leads to errors, when the ISO date is not passed as a valid date in the localization config.


### Checklist




#### QA


### Additional context

additional issues fixed by this commit:
https://github.com/Sage/carbon/issues/3945
https://github.com/Sage/carbon/issues/4453

### Testing instructions

1. run `npm start`
2. open `http://localhost:9001/?path=/story/date-input--default-story`
3. open `http://localhost:9001/?path=/story/date-input--with-custom-date-format`

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

https://codesandbox.io/s/friendly-burnell-muhwp?file=/src/index.js

Invariant Violation when a date with slashes is not added in valid date list:

https://codesandbox.io/s/hungry-goldstine-f7pip
